generator/structs.py

```diff
@@ -0,0 +1,48 @@
+'''
+Generate header files that set up structs for all buses.
+Run this file (with the spec path as a command line argument) to write just
+these files or main.py to write all files.
+'''
+import sys
+sys.path.append('ParseCAN')
+import ParseCAN
+from common import structs_path, coord, ifndef, endif
+
+def write(car, output_path=structs_path):
+    '''
+    Write the header files for the main structs in the library.
+
+    :param output_path: file to be written to
+    :param car: CAN spec
+    '''
+    header_name = '_CAN_LIBRARY_STRUCTS'
+
+    with open(output_path, 'w') as f:
+        fw = f.write
+
+        fw(ifndef(header_name))
+        fw('#include <stdint.h>\n')
+        fw('#include <stdbool.h>\n\n')
+        fw('#include "constants.h"\n\n')
+        fw('#include "enum_segments.h"\n\n')
+
+        for bus in car.buses:
+            for msg in bus.messages:
+                fw('typedef struct {\n')
+
+                for seg in msg.segments:
+                    if seg.c_type == 'enum':
+                        enum_name = coord(bus.name, msg.name, seg.name)
+                        fw('\t{} {};\n'.format(enum_name, seg.name))
```

Comment:
Need to add an _T to the end of this format string (example: can0_VcuToDash_limp_state_T is correct, but this code actually only inputs can0_VcuToDash_limp_state)